### PR TITLE
158595336

### PR DESCRIPTION
**Pull request by Darius-Ndubi**
## What does this PR do?

- User sign in endpoint to verify entered credentials.
- Adds un-signed up user rejection to sign in endpoint.

## What is the relevant PT story?
[#158595336](https://www.pivotaltracker.com/story/show/158595336)

### CODE REVIEW

**Comment by esirK** [comment]
Kindly push your other fixes so that I can review them.

**Comment by Darius-Ndubi** [comment]
okay let me do that just now

File: api/app/views.py [hunk 1]
```diff
@@ -10,19 +10,16 @@
...
-                ride_id = ride[detail]
-            elif detail == 'Title':
-                ride_title = ride[detail]
+        ride_id = ride.get('id')
+        ride_title = ride.get('title')
     
         #add the id and title to dictionary
-        All_rides.update({ride_id: ride_title})
-    return jsonify(All_rides)
+        all_rides.update({ride_id: ride_title})
```
Comment (esirK) [inline api/app/views.py]: according to the name `all_rides`, would someone expect to find rides with full details of just the ride title??

File: api/app/views.py [hunk 2]
```diff
@@ -100,15 +86,13 @@ def ride_request(id, requests):
...
-        for detail in ride:
-            if detail == 'id':
-                if ride[detail] == id:
-                    to_delete = ride
-                    #find the index of the ride
-                    ride_index = rides.index(to_delete)
+        if ride.get('id')==id:
+            to_delete = ride
+            #find the index of the ride
+            ride_index = rides.index(to_delete)
```
Comment (esirK) [inline api/app/views.py:73]: are you sure this will work??

File: api/app/views.py [hunk 3]
```diff
@@ -119,40 +103,36 @@ def edit_ride(id):
     """Loop through all rides and find ride with entered id"""
     if request.method=='GET':
         for ride in rides:
-            for detail in ride:
-                if ride[detail]==id:
-                    to_edit=ride
-                    return jsonify({"You want to edit":to_edit})
+            if ride.get('id') == id:
+                to_edit=ride
+                return jsonify({"You want to edit":to_edit})
```
Comment (esirK) [inline api/app/views.py:90]: is this necessary??

File: api/apidoc/app.py [hunk 4]
```diff
@@ -1,224 +1,179 @@
...
+        """# without schema
+        users.append(self.args)"""
+       
+        user = Users(id =self.args.get('id'), email=self.args.get(
+            'email'),password= self.args.get('password'), username=self.args.get('username'))
+
+        schema=UsersSchema()
+        result=schema.dumps(user)
+        print result
+        users.append(result)
```
Comment (esirK) [inline api/apidoc/app.py]: we agreed you should add a `user` object and not the `json` representation of the user object right??

File: api/apidoc/models.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        self.title=title
+        self.ride_date=ride_date
+        self.distance=distance
+        self.start_time=start_time
+        self.arrival_time=arrival_time
+        self.ride_price=ride_price
+        
+
+    
+    def addRide(self,ride_data):
```
Comment (esirK) [inline api/apidoc/models.py]: Who is responsible for managing rides??

File: api/apidoc/data.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
+rides = [
+    {
```
Comment (esirK) [inline api/apidoc/data.py:2]: Why not just leave this as an empty list??

**Review by wandesky — COMMENTED** [review]
I wrote this to illustrate to @Darius-Ndubi how comments are done in github

**Pull request merged**